### PR TITLE
update changelog for 2.072

### PR DESCRIPTION
**Pull request by 9il**

### CODE REVIEW

**Comment by 9il** [comment]
@CyberShadow tests failing `backend/optabgen.c:1079:1: fatal error: error writing to /tmp/cctr9ayP.s: No space left on device`

**Comment by CyberShadow** [comment]
Thx, looking at it

**Comment by CyberShadow** [comment]
Should be fixed, rebase (or push anything) to retest

File: changelog.dd [hunk 1]
```diff
@@ -73,6 +82,86 @@ assert(m.front[1] == "12");
...
+        allocation utilities were added to $(STDMODREF ndslice.slice, std.experimental.ndslice)
+        )
+    $(P Example: Transposing common 2D array using `ndslice`)
+-----
+import std.experimental.ndslice;
+
+auto ar = [[0, 1, 2], [3, 4, 5]];
+
+auto sh = ar.shape;         // [2, 3] type of size_t[2]
+auto sl = slice!int(sh);    // allocates slice with corresponding shape
```
Comment (wilzbach) [inline changelog.dd]: just `slice(sh)` works too ;-)

File: changelog.dd [hunk 2]
```diff
@@ -73,6 +82,86 @@ assert(m.front[1] == "12");
...
+$(LI $(LNAME2 slice_mio, partial support for Math Index Order was added to `std.experimental.ndslice.slice.Slice`.)
+---
+import std.experimental.ndslice;
+
+auto sl = iotaSlice(3, 4);
+
+assert(sl[ 2, 3 ] == 11); // D & C index order
+assert(sl[[2, 3]] == 11); // D & C index order
+assert(sl( 3, 2 ) == 11); // Math & Fortran index order
+assert(sl([3, 2]) == 11); // Math & Fortran index order
```
Comment (wilzbach) [inline changelog.dd]: I would reduce this to only to two lins (variadic version only )

File: changelog.dd [hunk 3]
```diff
@@ -73,6 +82,86 @@ assert(m.front[1] == "12");
...
+$(LI $(LNAME2 slice_ptr, `ptr` property and public constructor were added to
+    `std.experimental.ndslice.slice.Slice`.)
+    $(P `ptr` property allows to access `Slice`'s
+        underlaying pointer or range. It is suggested to read about
+        $(REF Slice, std,experimental,ndslice,slice)
+        internal binary representation before using the property.
+        `ptr` property is used in $(LINK2 https://github.com/libmir/mir, Mir)
+        for $(LINK2 http://docs.mir.dlang.io/latest/mir_sparse_package.html, sparse tensors).
+        `ndslice` developer mirror in Mir will be removed as
+        soon as LDC (LLVM D compiler) support DMDFE 2.072.
```
Comment (wilzbach) [inline changelog.dd]: support -> supports

**Comment by wilzbach** [comment]
@9il Looks very good to me - just had a couple of minor comments ;-)

File: changelog.dd (hunk 1 above)
Comment (9il) [inline changelog.dd]: > just ar.slice works too ;-)

no

**Comment by 9il** [comment]
fixed & rebased)

File: changelog.dd (hunk 1 above)
Comment (wilzbach) [inline changelog.dd]: oh sorry - that's right, but then we could fix that? 
Or is this problematic because `shape` will throw an exception?

Comment (9il) [inline changelog.dd]: No, this just looks like bad API

**Comment by wilzbach** [comment]
> No, this just looks like bad API

-> will submit a suggestion as PR - let's discuss there then.
To others: This is only about showing off some changes to the user ;-)

So everything good from side.

**Comment by 9il** [comment]
> -> will submit a suggestion as PR - let's discuss there then.
> To others: This is only about showing off some changes to the user ;-)
> 
> So everything good from my side.

I mean that creating a slice from ndarray is bad API. And we have not such approach.

File: changelog.dd [hunk 4]
```diff
@@ -73,6 +82,84 @@ assert(m.front[1] == "12");
...
+
+assert(ret.x.approxEqual(4.0));
+assert(ret.y.approxEqual(0.0));
+-------
+)
+
+$(LI $(LNAME2 slice_ptr, `ptr` property and public constructor were added to
+    `std.experimental.ndslice.slice.Slice`.)
+    $(P `ptr` property allows to access `Slice`'s
+        underlaying pointer or range. It is suggested to read about
```
Comment (JackStouffer) [inline changelog.dd]: > It is suggested to read about

This is somewhat awkward English. Something like "Please refer to `$(REF Slice, std,experimental,ndslice,slice)`'s internal binary representation before using the property." would sound more natural.

Comment (9il) [inline changelog.dd]: thanks!

File: changelog.dd [hunk 5]
```diff
@@ -73,6 +82,84 @@ assert(m.front[1] == "12");
...
+        $(REF Slice, std,experimental,ndslice,slice)
+        internal binary representation before using the property.
+        `ptr` property is used in $(LINK2 https://github.com/libmir/mir, Mir)
+        for $(LINK2 http://docs.mir.dlang.io/latest/mir_sparse_package.html, sparse tensors).
+        `ndslice` developer mirror in Mir will be removed as
+        soon as LDC (LLVM D compiler) supports DMDFE 2.072.
```
Comment (JackStouffer) [inline changelog.dd]: > DMDFE

I've never seen this abbreviated like this before, and it's kind of "[inside baseball](https://en.wikipedia.org/wiki/Inside_baseball_(metaphor))". I think something like "as soon as LDC (LLVM D compiler) supports D version 2.072." would be clearer.

**Comment by JackStouffer** [comment]
I don't know why my second comment was marked as outdated, but once it's fixed this looks good to go.

**Comment by wilzbach** [comment]
> but once it's fixed this looks good to go.

Let's go?

**Comment by JackStouffer** [comment]
> Let's go?

Would if I had merge rights ;)

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Comment by DmitryOlshansky** [comment]
Needs rebase

**Pull request merged**